### PR TITLE
Make Courant absolute time aware

### PR DESCRIPTION
**Pull request by charleskawczynski**
# Description

This PR makes the Courant calculations time-aware. As pointed out by @akshaysridhar, this is useful if we want to make `C_smag` (or, effectively, the inverse timestep) decay so that we can speed up calculations after initial transient stiffness. It also simplifies some of the Courant functions.

# For review by CLIMA developers

- [x] There are no open pull requests for this already
- [x] CLIMA developers with relevant expertise have been assigned to review this submission
- [x] The code conforms to the [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by bischtob** [comment]
Does the formatter also have variable naming conventions :D?

**Comment by charleskawczynski** [comment]
> Does the formatter also have variable naming conventions :D?

Nope :P

**Comment by blallen** [comment]
I like this and stole it for #841 :) good job @charleskawczynski

File: src/Atmos/Model/courant.jl [hunk 1]
```diff
@@ -55,21 +57,7 @@ function diffusive_courant(
     direction,
 )
     ν, τ = turbulence_tensors(m.turbulence, state, diffusive, aux, 0)
```
Comment (akshaysridhar) [inline src/Atmos/Model/courant.jl]:
```suggestion
    ν, τ = turbulence_tensors(m.turbulence, state, diffusive, aux, t)
```

Comment (charleskawczynski) [reply src/Atmos/Model/courant.jl]: Looks good, I’ll update this!

**Comment by charleskawczynski** [comment]
@akshaysridhar added time as a parameter here!

**Comment by kpamnany** [comment]
Squash and r+?

**Comment by kpamnany** [comment]
Allowing PRs that reduce CI runtime to go ahead.

**Pull request merged**